packages/cli-kit/src/private/node/session.ts

```diff
@@ -101,6 +101,28 @@ export interface OAuthSession {
   userId: string
 }
 
+let userId: undefined | string
+
+/**
+ * Retrieves the user ID from the current session or returns 'unknown' if not found.
+ * This function first checks for a cached user ID in memory (obtained in the current run)
+ * Then attempts to fetch it from the secure store. (from a previous auth session)
+ * If no user ID is found, it returns 'unknown'.
+ *
+ * @returns A Promise that resolves to the user ID as a string.
+ */
+export async function getLastSeenUserIdAfterAuth(): Promise<string> {
+  if (userId) return userId
+  const currentSession = (await secureStore.fetch()) || {}
+  const fqdn = await identityFqdn()
+  const cachedUserId = currentSession[fqdn]?.identity.userId
```

Comment:
This should never crash no?
`currentSession` might be undefined and that's handled
If it's not undefined, there is an identity object, with a `userId` value that might be a `string` or `undefined`, but that's ok, that's when we return "unknown" 

This will happen:
- For users running the code before caching any userId (probably just after updating the CLI)
- When running a CLI command like `build` without any cached authentication (after a logout)